gpflow/kernels.py

```diff
@@ -424,13 +424,28 @@ class Matern32(Stationary):
     The Matern 3/2 kernel
     """
 
+    # Used to bypass gradients of tf.sqrt that can lead to infs.
+    tf.RegisterGradient("Matern32Grad")(
+        lambda op, grad: grad * .5 * tf.exp(op.inputs[0])
+    )
+
     @params_as_tensors
     def K(self, X, X2=None, presliced=False):
         if not presliced:
             X, X2 = self._slice(X, X2)
-        r = self.scaled_euclid_dist(X, X2)
-        return self.variance * (1. + np.sqrt(3.) * r) * \
-               tf.exp(-np.sqrt(3.) * r)
+        d = self.square_dist(X, X2)
+
+        with self.graph.gradient_override_map(
```

Comment:
A bit more about *gradient_override_map*
From docs: https://www.tensorflow.org/api_docs/python/tf/Graph

gradient_override_map(op_type_map)
**EXPERIMENTAL**: A context manager for overriding gradient functions.

Reply:
Does it mean there is no guarantee that this will work or supported in future by TensorFlow?

Reply:
Yeah, I guess this might not be future proof.